static/app/components/nav/sidebar.tsx

```diff
@@ -89,37 +91,79 @@ const SidebarItemList = styled('ul')`
   }
 `;
 
-function SidebarItem({item}: {item: NavSidebarItem}) {
-  const location = useLocation();
-  const isActive = isNavItemActive(item, location);
-  const isSubmenuActive = isSubmenuItemActive(item, location);
-  const _to = resolveNavItemTo(item);
-  const linkProps = _to ? makeLinkPropsFromTo(_to) : {to: '#'};
+interface SidebarItemProps {
+  item: NavSidebarItem;
+}
+
+function SidebarItem({item}: SidebarItemProps) {
+  const to = resolveNavItemTo(item);
+  const Component = to ? SidebarLink : SidebarMenu;
 
   const FeatureGuard = item.feature ? Feature : Fragment;
   const featureGuardProps: any = item.feature ?? {};
 
   return (
     <FeatureGuard {...featureGuardProps}>
       <SidebarItemWrapper>
-        <SidebarLink
-          {...linkProps}
-          aria-current={isActive ? 'page' : undefined}
-          aria-selected={isActive || isSubmenuActive}
-        >
-          <InteractionStateLayer hasSelectedBackground={isActive || isSubmenuActive} />
+        <Component item={item} key={item.label}>
           {item.icon}
           <span>{item.label}</span>
-        </SidebarLink>
+        </Component>
       </SidebarItemWrapper>
     </FeatureGuard>
   );
 }
 
-const SidebarLink = styled(Link)`
+const StyledLink = styled(Link)`
   position: relative;
 `;
 
+const StyledButton = styled('button')`
+  border: none;
+  position: relative;
+  background: transparent;
+  min-width: 58px;
+
+  ${linkStyles}
+`;
+
+function SidebarLink({children, item}: PropsWithChildren<SidebarItemProps>) {
+  const location = useLocation();
+  const isActive = isNavItemActive(item, location);
+  const isSubmenuActive = isSubmenuItemActive(item, location);
+  const to = resolveNavItemTo(item);
+  const linkProps = makeLinkPropsFromTo(to!);
+
+  return (
+    <StyledLink
+      {...linkProps}
+      className={isActive || isSubmenuActive ? 'active' : undefined}
+      aria-current={isActive ? 'page' : undefined}
+    >
+      <InteractionStateLayer hasSelectedBackground={isActive || isSubmenuActive} />
+      {children}
+    </StyledLink>
+  );
+}
+
+function SidebarMenu({item}: PropsWithChildren<SidebarItemProps>) {
+  return (
+    <DropdownMenu
+      position="right-end"
+      trigger={(props, isOpen) => {
+        return (
+          <StyledButton {...props}>
+            <InteractionStateLayer hasSelectedBackground={isOpen} />
+            {item.icon}
+            <span>{item.label}</span>
```

Comment:
Shouldn't this just render `{children}` since we pass these two things in the component above

Reply:
Good catch, refactored in 9580e0c83f2!